### PR TITLE
Add new monitor type

### PR DESCRIPTION
**Pull request by SebastianIoanPop**

### CODE REVIEW

**Comment by SebastianIoanPop** [comment]
Included the other PR with updated links

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Redis StatsD Reporter [![Build Status](https://travis-ci.org/andyroyle/redis-statsd-reporter.svg?branch=master)](https://travis-ci.org/andyroyle/redis-statsd-reporter)
+Redis StatsD Reporter [![Build Status](https://travis-ci.org/ve-interactive/redis-statsd-reporter.svg?branch=master)](https://travis-ci.org/ve-interactive/redis-statsd-reporter)
```
Comment (nicolaspujol) [inline README.md]: Remove trailling space

File: lib/redis-client-factory.js [hunk 2]
```diff
@@ -3,7 +3,7 @@
 const Redis = require('ioredis');
 const util = require('util');
 const Async = require('async');
-
+
```
Comment (nicolaspujol) [inline lib/redis-client-factory.js]: Remove trailling spaces

File: lib/redis-key-counter.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    count: (cl, callback) => {
+        if (cl.keyCounters && cl.keyCounters.zset) {
+            cl.keyCounters.zset.forEach((key) => {
+                cl.zcount([key, '-inf', 'inf'], (err, res) => {
+                    callback(err, { key: key, count: res });
+                });
+            });
+        }
+    }
+};
```
Comment (nicolaspujol) [inline lib/redis-key-counter.js]: Add new line

File: lib/redis-key-counter.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+module.exports = {
+    count: (cl, callback) => {
+        if (cl.keyCounters && cl.keyCounters.zset) {
+            cl.keyCounters.zset.forEach((key) => {
```
Comment (nicolaspujol) [inline lib/redis-key-counter.js]: No need for ( ) around key:

(key => {

File: lib/redis-key-counter.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+                cl.zcount([key, '-inf', 'inf'], (err, res) => {
+                    callback(err, { key: key, count: res });
```
Comment (nicolaspujol) [inline lib/redis-key-counter.js]: You can simplify key: key by key `{ key, count: res }`

File: redis.json [hunk 6]
```diff
@@ -3,6 +3,9 @@
...
     "prefix": "my.redis.yay",
     "cluster": false,
-    "nodeNames": "prefix"
+    "nodeNames": "prefix",
+    "keyCounters": {
+      "zset": [ "sampleKey" ]
+    }
   }
-]
+]
```
Comment (nicolaspujol) [inline redis.json]: Add newline

File: tests/redis-key-counter.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+var should = require('should');
+
+describe('redis-key-counter', () => {
+    var keyCounter = require('../lib/redis-key-counter');
+
+    describe('should return client result', () => {
+        var fakeClient = {};
```
Comment (nicolaspujol) [inline tests/redis-key-counter.js]: Empty line after var declaration

File: tests/redis-key-counter.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+        fakeClient.zcount = (arr, cb) => {
+            cb(null, 2);
+        };
+
+        keyCounter.count(fakeClient, (err, res) => {
+            res.key.should.be.exactly('temp');
+            res.count.should.be.exactly(2);
+        });
+    });
+});
```
Comment (nicolaspujol) [inline tests/redis-key-counter.js]: Add new line

File: index.js [hunk 9]
```diff
@@ -5,6 +5,7 @@ const util = require('util');
 const infoParser = require('./lib/info-parser');
 const redisClientFactory = require('./lib/redis-client-factory');
 const libUtils = require('./lib/utils');
+var keyCounter = require('./lib/redis-key-counter');
```
Comment (nicolaspujol) [inline index.js]: You can use let instead of var

File: index.js [hunk 10]
```diff
@@ -34,7 +35,18 @@ redisClients.forEach((c) => {
...
+    keyCounter.count(c, (err, stat) => {
+      if (err) {
+        util.log(`[${c.host}] ${err}`);
+        return;
+      }
+
+      statsdClient.gauge(`${c.prefix}.${stat.key}.${c.suffix}`, stat.count);
+    });
+
   }, (statsConfig.interval || 10) * 1000);
```
Comment (nicolaspujol) [inline index.js]: Magic numbers: 10 and 1000 should be stored in const/let with a name

File: tests/redis-key-counter.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+var should = require('should');
```
Comment (nicolaspujol) [inline tests/redis-key-counter.js]: You should use let instead of var

Comment (nicolaspujol) [inline tests/redis-key-counter.js]: (here and after)

File: tests/redis-key-counter.js [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+
+describe('redis-key-counter', () => {
+    var keyCounter = require('../lib/redis-key-counter');
+
+    describe('should return client result', () => {
+        var fakeClient = {};
+        fakeClient.keyCounters = { zset: [ 'temp' ] };
+        fakeClient.zcount = (arr, cb) => {
+            cb(null, 2);
```
Comment (nicolaspujol) [inline tests/redis-key-counter.js]: 2 is a magic number

File: index.js [hunk 13]
```diff
@@ -34,7 +36,18 @@ redisClients.forEach((c) => {
...
+    keyCounter.count(c, (err, stat) => {
+      if (err) {
+        util.log(`[${c.host}] ${err}`);
+        return;
+      }
+
+      statsdClient.gauge(`${c.prefix}.${stat.key}.${c.suffix}`, stat.count);
+    });
+
+  }, (statsConfig.interval || ten) * milisecondToSecondMultiplier);
```
Comment (nicolaspujol) [inline index.js]: I would rename `ten` to defaultInterval

Comment (nicolaspujol) [inline index.js]: I would rename `milisecondToSecondMultiplier` to `oneSecInMs`

File: tests/redis-key-counter.js [hunk 14]
```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+let should = require('should');
+let exactKeyCount = 2;
+
+describe('redis-key-counter', () => {
+    let keyCounter = require('../lib/redis-key-counter');
+
+    describe('should return client result', () => {
+        var fakeClient = {};
```
Comment (nicolaspujol) [inline tests/redis-key-counter.js]: Use let

File: tests/redis-key-counter.js [hunk 15]
```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+let should = require('should');
+let exactKeyCount = 2;
```
Comment (nicolaspujol) [inline tests/redis-key-counter.js]: Can this name be a bit more descriptive ?

**Pull request merged**